spec/models/mixins/supports_feature_mixin_spec.rb

```diff
@@ -0,0 +1,96 @@
+describe SupportsFeatureMixin do
+  class Post
+    module Operations
+      include SupportsFeatureMixin
+      supports :publish
+      supports :archive
+      supports_not :fake, "We keep it real!"
+    end
+    include Operations
+
+    module A
+      module B
+        module C
+          include SupportsFeatureMixin
+          supports :nested
+        end
+        include C
+      end
+      include B
+    end
+    include A
+  end
+
+  class SpecialPost < Post
+    attr_accessor :frontpage, :bribe
+
+    def initialize(options = {})
+      self.frontpage = options[:frontpage]
+      self.bribe = options[:bribe]
+    end
+
+    module Operations
+      include SupportsFeatureMixin
+      supports :archive do |post|
+        post.unsupported[:archive] = "It's on the frontpage" if post.frontpage
+      end
+
+      supports :fake do |post|
+        post.unsupported[:fake] = "Need more money" unless post.bribe
+      end
+    end
+    include Operations
+  end
+
+  it "creates a supports_feature? method" do
+    expect(Post.supports_publish?).to be true
+  end
+
+  it "creates a support_feature? method" do
+    expect(Post.new.supports_publish?).to be true
+  end
+
+  it "subclasses instance level" do
+    expect(SpecialPost.new.supports_publish?).to be true
+  end
+
+  it "supports on class level" do
+    expect(SpecialPost.supports_archive?).to be true
+  end
+
+  it "is unsupported on condition" do
+    expect(SpecialPost.new(:frontpage => true).supports_archive?).to be false
+  end
+
+  it "is supported when key is not set" do
+    expect(SpecialPost.new.supports_archive?).to be true
+  end
+
+  it "returns the reason when accessing a message" do
+    frontpage_post = SpecialPost.new(:frontpage => true)
+    expect(frontpage_post.supports_archive?).to be false
+    expect(frontpage_post.unsupported[:archive]).to eq "It's on the frontpage"
+  end
+
+  it "returns the reason even when not asking for supported? first" do
+    expect(SpecialPost.new(:frontpage => true).unsupported[:archive]).to eq "It's on the frontpage"
+  end
+
+  it "returns false for fake" do
+    expect(Post.supports_fake?).to be false
+  end
+
+  it "returns a reason on an unsupported class method" do
+    expect(Post.unsupported[:fake]).to eq "We keep it real!"
+  end
+
+  it "returns true on the subclass" do
+    expect(SpecialPost.supports_fake?).to be true
+  end
+
+  it "propagates deeply nested supports" do
+    expect(Post.supports_nested?).to be true
```

Comment:
this one is failing, because I dont propagate up deeply nested includes.

@blomquisg @Fryguy @chessbyte do we want to keep the current codebase layout of nested modules?
If I'd put all `supports` methods into either the class `Vm` or one shared module `VmOperations` this would work.